azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/AbstractKeyVaultTemplate.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.keyvault.spring;
+
+import com.microsoft.azure.AzureResponseBuilder;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.serializer.AzureJacksonAdapter;
+import com.microsoft.azure.spring.support.UserAgent;
+import com.microsoft.azure.utils.Memoizer;
+import com.microsoft.rest.RestClient;
+
+import java.util.function.Function;
+
+import static com.microsoft.azure.keyvault.spring.Constants.DEFAULT_REFRESH_INTERVAL_MS;
+
+public abstract class AbstractKeyVaultTemplate implements KeyVaultOperation {
+    private static final String KEY_VAULT_URL = "https://%s.vault.azure.net/";
+
+    protected final AzureKeyVaultCredential credential;
+    protected Function<String, KeyVaultClient> keyVaultClientCreator = keyVaultClientCreator();
+    private boolean allowTelemetry = false;
+    protected long refreshIntervalMS = DEFAULT_REFRESH_INTERVAL_MS;
```

Comment:
I suggest to have kind of option to disable periodic refresh. I would say in most cases, secret won't change. Maybe if this property is not provided, no refresh.

Reply:
With default refresh interval(-1), it will not be refreshed.